From 0157ea26db33e55c0087b99f3da4f78a1cc777a1 Mon Sep 17 00:00:00 2001
From: Yinxin Sun <yinxsun@qti.qualcomm.com>
Date: Thu, 6 Nov 2025 17:56:49 +0800
Subject: [PATCH] device: Handle wifi device as non-external

Mark out external lines to handle wifi as non-external.

Signed-off-by: Yinxin Sun <yinxsun@qti.qualcomm.com>
---
 device.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/device.c b/device.c
index b1213a0..fe934eb 100644
--- a/device.c
+++ b/device.c
@@ -801,9 +801,9 @@ static struct device *
 device_create_default(const char *name, bool external)
 {
 	struct device *dev;
-
-	if (!external && system_if_force_external(name))
-		return NULL;
+	// netifd mark wifi device as external, but we handle wifi as non-external, so mark out below line
+	// if (!external && system_if_force_external(name))
+	//	return NULL;
 
 	D(DEVICE, "Create simple device '%s'\n", name);
 	dev = calloc(1, sizeof(*dev));
-- 
2.34.1

